### PR TITLE
Employee time slots

### PR DESCRIPTION
**Pull request by RussellSk**

### CODE REVIEW

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/EmployeeAvailabilityTimeService.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.exadel.team2.sandbox.service;
+
+import com.exadel.team2.sandbox.web.employee_availability_time.CreateEmployeeAvailabilityTimeDto;
+import com.exadel.team2.sandbox.web.employee_availability_time.ResponseEmployeeAvailabilityTimeDto;
+import com.exadel.team2.sandbox.web.employee_availability_time.UpdateEmployeeAvailabilityTimeDto;
+
+public interface EmployeeAvailabilityTimeService extends GeneralService<ResponseEmployeeAvailabilityTimeDto,
```
Comment (ikantsevich) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/EmployeeAvailabilityTimeService.java:7]: Do we need to add two methods?
getByEmployeeId(Long id) 
 deleteAllByEmployeeId(Long employeeId)

File: sandbox-backend/src/main/resources/db/changelog/changes/20210505_create_table_employee_availability_time.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                  name: EVAT_DATETIME
+                  type: DATETIME
+                  constraints:
+                    nullable: false
+              - column:
+                  name: EVAT_CREATED_AT
+                  type: DATETIME
+                  defaultValue: now()
+                  constraints:
+                    nullable: true
```
Comment (ikantsevich) [inline sandbox-backend/src/main/resources/db/changelog/changes/20210505_create_table_employee_availability_time.yaml:36]: How EVAT_CREATED_AT can be null?

File: sandbox-backend/src/main/resources/db/changelog/changes/20210505_create_table_employee_availability_time.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+              - column:
+                  name: EVAT_UPDATED_AT
+                  type: DATETIME
+                  constraints:
+                    nullable: true
```
Comment (ikantsevich) [inline sandbox-backend/src/main/resources/db/changelog/changes/20210505_create_table_employee_availability_time.yaml:41]: The same

File: sandbox-web/src/main/java/com/exadel/team2/sandbox/controller/EmployeeController.java [hunk 4]
```diff
@@ -20,6 +24,7 @@
 public class EmployeeController {
 
     private final EmployeeService employeeService;
+    private final EmployeeAvailabilityTimeServiceImpl employeeAvailabilityTimeService;
```
Comment (ikantsevich) [inline sandbox-web/src/main/java/com/exadel/team2/sandbox/controller/EmployeeController.java:27]: I think we should inject interface here

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/EmployeeAvailabilityTimeService.java (hunk 1 above)
Comment (RussellSk) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/EmployeeAvailabilityTimeService.java:7]: Yes we need these methods. getByEmployeeId(Long id) returns all time slots that belongs to EmployeeId.

**Review by ikantsevich — APPROVED** [review]

**Pull request merged**